parsl/data_provider/data_manager.py

```diff
@@ -149,30 +149,30 @@ def stage_in(self, file, executor):
 
         if file.scheme == 'ftp':
             working_dir = self.dfk.executors[executor].working_dir
-            stage_in_app = self._ftp_stage_in_app(executor=executor)
+            stage_in_app = self._ftp_stage_in_app(executor=executor, dfk=self.dfk)
```

Comment:
Would we ever call this with `dfk != self.dfk`? If not, maybe we should remove the dfk keyword argument altogether and hardcode it the method definition?

Reply:
I think that would be a reasonable change -- I'll make a commit.